File: app/src/main/java/me/jfenn/alarmio/adapters/SimplePagerAdapter.java
package me.jfenn.alarmio.adapters;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.Objects;

import me.jfenn.alarmio.fragments.BasePagerFragment;
import me.jfenn.alarmio.interfaces.FragmentInstantiator;


public class SimplePagerAdapter extends FragmentStatePagerAdapter {

    private final FragmentInstantiator[] fragments;

    public SimplePagerAdapter(Context context, FragmentManager fm, FragmentInstantiator... fragments) {
        super(fm);
        this.fragments = fragments;
    }

    @NonNull
    @Override
    public BasePagerFragment getItem(int position) {
        return Objects.requireNonNull(fragments[position].newInstance(position));
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return fragments[position].getTitle(position);
    }

    @Override
    public int getCount() {
        return fragments.length;
    }

}

File: app/src/main/java/me/jfenn/alarmio/adapters/SimplePagerAdapter.kt
package me.jfenn.alarmio.adapters

import android.content.Context

import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

import me.jfenn.alarmio.fragments.BasePagerFragment
import me.jfenn.alarmio.interfaces.FragmentInstantiator


class SimplePagerAdapter(
    context: Context?,
    fm: FragmentManager?,
    vararg fragments: FragmentInstantiator
) : FragmentStatePagerAdapter(
    fm!!
) {
    private val fragments: Array<out FragmentInstantiator>

    init {
        this.fragments = fragments
    }

    override fun getItem(position: Int): BasePagerFragment {
        return fragments[position].newInstance(position)!!
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return fragments[position].getTitle(position)
    }

    override fun getCount(): Int {
        return fragments.size
    }
}
